pilot/pkg/config/monitor/monitor.go

```diff
@@ -91,11 +92,16 @@ func (m *Monitor) checkAndUpdate() {
 			m.createConfig(newConfig)
 			newIndex++
 		} else {
+			oldConfig.Mutex = &sync.RWMutex{}
```

Comment:
These are two different locks here

Reply:
Not sure why you think those are different locks. I ran the racetest in a loop 30 times with no failures so if they were different this should have failed on the first runs.

Reply:
This is mutex 1

Reply:
>I ran the racetest in a loop 30 times with no failures

I think this because a later initiated mutex takes effect